File: CS/NavigationScrolling/Converters.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Controls;
using System.Windows.Data;

namespace DXSample {
    public class OrientationInverter : IValueConverter {
        public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture) {
            Orientation inputValue = (Orientation)value;
            switch (inputValue) {
                case Orientation.Horizontal:
                    return Orientation.Vertical;
                default:
                    return Orientation.Horizontal;
            }
        }

        public object ConvertBack(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture) {
            Orientation inputValue = (Orientation)value;
            switch (inputValue) {
                case Orientation.Horizontal:
                    return Orientation.Vertical;
                default:
                    return Orientation.Horizontal;
            }
        }
    }
}

File: VB/NavigationScrolling/Converters.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Controls
Imports System.Windows.Data

Namespace DXSample
	Public Class OrientationInverter
		Implements IValueConverter
		Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.Convert
			Dim inputValue As Orientation = CType(value, Orientation)
			Select Case inputValue
				Case Orientation.Horizontal
					Return Orientation.Vertical
				Case Else
					Return Orientation.Horizontal
			End Select
		End Function

		Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.ConvertBack
			Dim inputValue As Orientation = CType(value, Orientation)
			Select Case inputValue
				Case Orientation.Horizontal
					Return Orientation.Vertical
				Case Else
					Return Orientation.Horizontal
			End Select
		End Function
	End Class
End Namespace
